ecs-cli/modules/cli/local/network/network.go

```diff
@@ -0,0 +1,217 @@
+		IPAM: &network.IPAM{
+			Config: []network.IPAMConfig{
+				{
+					Subnet: ecsLocalNetworkSubnet,
+				},
+			},
+		},
+	})
+	if err != nil {
+		logrus.Fatalf("Failed to create network %s with subnet %s due to %v", EcsLocalNetworkName, ecsLocalNetworkSubnet, err)
+	}
+	logrus.Infof("Created network %s with ID %s", EcsLocalNetworkName, resp.ID)
+}
+
+// setupLocalEndpointsContainer creates the Local Endpoints container if one doesn't exist already.
+func setupLocalEndpointsContainer(docker idempotentContainerStarter) {
+	containerID := createLocalEndpointsContainer(docker)
+	startContainer(docker, containerID)
+}
+
+// createLocalEndpointsContainer returns the ID of the newly created container.
+// If the container already exists, returns the ID of the existing container.
+func createLocalEndpointsContainer(docker idempotentContainerStarter) string {
+	ctx, cancel := context.WithTimeout(context.Background(), dockerTimeout)
+	defer cancel()
+
+	// See First Scenario in https://aws.amazon.com/blogs/compute/a-guide-to-locally-testing-containers-with-amazon-ecs-local-endpoints-and-docker-compose/
+	// for an explanation of these fields.
+	resp, err := docker.ContainerCreate(ctx,
+		&container.Config{
+			Image: localEndpointsImageName,
+			Env: []string{
+				"AWS_PROFILE=default",
+				"HOME=/home",
+			},
+		},
+		&container.HostConfig{
+			Binds: []string{
+				"/var/run:/var/run",
+				fmt.Sprintf("%s/.aws/:/home/.aws/", os.Getenv("HOME")),
+			},
+		},
+		&network.NetworkingConfig{
+			EndpointsConfig: map[string]*network.EndpointSettings{
+				EcsLocalNetworkName: {
+					NetworkID: EcsLocalNetworkName,
+					IPAMConfig: &network.EndpointIPAMConfig{
+						IPv4Address: localEndpointsContainerIpAddr,
+					},
+				},
+			},
+		},
+		localEndpointsContainerName,
+	)
+	if err != nil {
+		if strings.Contains(err.Error(), "Conflict") {
+			// We already created this container before since there is a name conflict, fetch its ID and return it.
+			containerID := localEndpointsContainerID(docker)
+			logrus.Infof("The %s container already exists with ID %s", localEndpointsContainerName, containerID)
+			return containerID
+		}
+		logrus.Fatalf("Failed to create container %s due to %v", localEndpointsContainerName, err)
+	}
+
+	logrus.Infof("Created the %s container with ID %s", localEndpointsContainerName, resp.ID)
+	return resp.ID
+}
+
+// localEndpointsContainerID returns the ID of the Local Endpoints container.
+func localEndpointsContainerID(docker idempotentContainerStarter) string {
+	ctx, cancel := context.WithTimeout(context.Background(), dockerTimeout)
+	defer cancel()
+
+	resp, err := docker.ContainerList(ctx, types.ContainerListOptions{
+		All: true,
+		Filters: filters.NewArgs(
+			filters.Arg("name", localEndpointsContainerName),
+		),
+	})
+	if err != nil {
+		logrus.Fatalf("Failed to list containers with name %s due to %v", localEndpointsContainerName, err)
+	}
+	return resp[0].ID
```

Comment:
I think you should add a check that the response is non-nil and contains at least one element.

Also Nit: For cases like this (where you're using a list API but you expect one result), I like to explicitly return an error if the response has more than one element. Just in case something goes really weird. May be I'm just paranoid though.